### PR TITLE
gourmand: init at 1.0.0, +dependency 

### PR DESCRIPTION
**Pull request by DeeUnderscore**
###### Motivation for this change
Gourmand is a fork of the Gourmet Recipe Manager, a desktop cookbook/recipe manager. It is actively maintained, and runs under Python 3 (Gourmet used Python 2).

pythonPackages.ebooklib, also in this PR, is the library that Gourmet uses for exporting recipes to epub.
 
Website scraping plugins do not work in this derivation, as they'd need another dependency not currently in Nixpkgs, and they are gone in the latest release candidate for the next version of Gourmand, replaced instead by the recipe-scrapers library (which I managed to build, but omitted from this PR). 

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pbsds** [comment]
Please rebase this on top of master

```shell
Auto-merging pkgs/top-level/python-packages.nix
CONFLICT (content): Merge conflict in pkgs/top-level/python-packages.nix
Automatic merge failed; fix conflicts and then commit the result.
```

File: pkgs/development/python-modules/ebooklib/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  version = "0.17.1";
+
+  src = fetchPypi {
+    inherit version;
+    pname = "EbookLib";
+    sha256 = "sha256-/iPiLCgFAZbGjbPnsTsle/OUJtknyzlcbyzBOsETJ/E=";
+  };
+
+  propagatedBuildInputs = [ lxml six ];
+
```
Comment (onny) [inline pkgs/development/python-modules/ebooklib/default.nix:19]: Maybe you could explicitly enforce tests by using `checkInputs = [ pytestCheckHook ];` and also use `pythonImportsCheck = [ "ebooklib"];`

**Comment by DeeUnderscore** [comment]
Rebased on master

File: pkgs/development/python-modules/ebooklib/default.nix (hunk 1 above)
Comment (DeeUnderscore) [reply pkgs/development/python-modules/ebooklib/default.nix:19]: The package seems to have no tests at all, so the pytest hook wouldn't really help. Added the import check, though.

**Comment by DeeUnderscore** [comment]
When rebased on master, 1.0.0 seems to no longer work with the sqlalchemy now in Nixpkgs. 

I'm not really interested in continuing to work on this, though, so closing.

**Comment by altsalt** [comment]
What timing, I have just begun using NixOS and was wondering what it would take to get gourmand packaged... Would it make more sense to reopen this or start a new PR? Suggestions about where to look to overcome the place you got stuck?

**Comment by pbsds** [comment]
In such cases I copy/continue from their work in a new PR, [tagging the original author as a co-author](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors#creating-co-authored-commits-on-the-command-line), and offerer shared package maintainership.

**Comment by DeeUnderscore** [comment]
> What timing, I have just begun using NixOS and was wondering what it would take to get gourmand packaged... Would it make more sense to reopen this or start a new PR? Suggestions about where to look to overcome the place you got stuck?

I haven't looked at the details closely, but it's likely upstream would have to be patched to support SQLAlchemy >2.0. This might be achievable while retaining upstream's current SQLAlchemy 1.4 compatibility, so I'd check if that's possible, send whatever patches to upstream, and then, for the Nixpkgs package, probably target the latest Gourmand RC release, with the patches applied. The RCs have a different set of dependencies, but it looks like a lot of them are packaged now.

**Comment by AndrewKvalheim** [comment]
I’ve been able to run Gourmand by just providing SQLAlchemy 1.4 again: https://github.com/NixOS/nixpkgs/compare/master...AndrewKvalheim:nixpkgs:gourmand

It looks like several packages do the same—

- [buildbot](https://github.com/NixOS/nixpkgs/blob/d76e8633bfe92a60e59aa1b4074c65f0bee436a9/pkgs/development/tools/continuous-integration/buildbot/default.nix#L9-L20)
- [calibre-web](https://github.com/NixOS/nixpkgs/blob/d76e8633bfe92a60e59aa1b4074c65f0bee436a9/pkgs/servers/calibre-web/default.nix#L11-L22)
- [ceph](https://github.com/NixOS/nixpkgs/blob/d76e8633bfe92a60e59aa1b4074c65f0bee436a9/pkgs/tools/filesystems/ceph/default.nix#L168-L179)
- [csvkit](https://github.com/NixOS/nixpkgs/blob/d76e8633bfe92a60e59aa1b4074c65f0bee436a9/pkgs/tools/text/csvkit/default.nix#L9-L20)
- [ntfy](https://github.com/NixOS/nixpkgs/blob/d76e8633bfe92a60e59aa1b4074c65f0bee436a9/pkgs/tools/misc/ntfy/default.nix#L20-L32)
- [pyca](https://github.com/NixOS/nixpkgs/blob/d76e8633bfe92a60e59aa1b4074c65f0bee436a9/pkgs/applications/video/pyca/default.nix#L13-L25)

—so maybe there’s a need for a shared `sqlalchemy_1`?

**Pull request closed without merging**